pycheribuild/projects/syzkaller.py

```diff
@@ -95,15 +106,19 @@ def needs_configure(self) -> bool:
 
     def compile(self, **kwargs):
         cflags = self.default_compiler_flags + self.default_ldflags
-
-        self.make_args.set_env(
-            HOSTARCH="amd64",
-            TARGETARCH=self.crosscompile_target.cpu_architecture.value,
-            TARGETOS="freebsd",
-            GOROOT=self.goroot.expanduser(),
-            GOPATH=self.gopath.expanduser(),
-            CC=self.CC, CXX=self.CXX,
-            PATH=self._new_path)
+        print(self._new_path)
+        args = {
+            "HOSTARCH":"amd64",
+            "TARGETARCH":"arm64" if self.crosscompile_target.cpu_architecture.value == "aarch64"
+                else self.crosscompile_target.cpu_architecture.value,
+            "TARGETOS":"freebsd",
+            "CC":self.CC, "CXX":self.CXX,
+            "PATH":self._new_path
+        }
+        if self.use_go == GOROOTOptions.UPSTREAM:
+            args["GOROOT"] = self.goroot.expanduser()
+            args["GOPATH"] = self.gopath.expanduser()
+        self.make_args.set_env(**args)
```

Comment:
We duplicate args in `clean()`. Instead of that, we should introduce `_setup_make_args()` as in `pycheribuild/projects/cross/cheribsd.py`, set args there and call `_setup_make_args()` in `compile()` and `clean()`.